managed/services/management/backup/backups_service_test.go

```diff
@@ -71,57 +80,97 @@ func setup(t *testing.T, q *reform.Querier, serviceName string) *models.Agent {
-	backupSvc := NewBackupsService(db, backupService, nil)
-	agent := setup(t, db.Querier, t.Name())
+	t.Run("mysql", func(t *testing.T) {
+		backupService := &mockBackupService{}
+		sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+		db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+		backupSvc := NewBackupsService(db, backupService, nil)
+		agent := setup(t, db.Querier, models.MySQLServiceType, t.Name())
+
+		for _, tc := range []struct {
+			testName    string
+			backupError error
+			code        backupv1beta1.ErrorCode
+		}{
+			{
+				testName:    "xtrabackup not installed",
+				backupError: backup.ErrXtrabackupNotInstalled,
+				code:        backupv1beta1.ErrorCode_ERROR_CODE_XTRABACKUP_NOT_INSTALLED,
+			},
+			{
+				testName:    "invalid xtrabackup",
+				backupError: backup.ErrInvalidXtrabackup,
+				code:        backupv1beta1.ErrorCode_ERROR_CODE_INVALID_XTRABACKUP,
+			},
+			{
+				testName:    "incompatible xtrabackup",
+				backupError: backup.ErrIncompatibleXtrabackup,
+				code:        backupv1beta1.ErrorCode_ERROR_CODE_INCOMPATIBLE_XTRABACKUP,
+			},
+		} {
+			t.Run(tc.testName, func(t *testing.T) {
+				backupError := fmt.Errorf("error: %w", tc.backupError)
+				backupService.On("PerformBackup", mock.Anything, mock.Anything).
+					Return("", backupError).Once()
+				ctx := context.Background()
+				resp, err := backupSvc.StartBackup(ctx, &backupv1beta1.StartBackupRequest{
+					ServiceId:     *agent.ServiceID,
+					LocationId:    "locationID",
+					Name:          "name",
+					Description:   "description",
+					RetryInterval: nil,
+					Retries:       0,
+				})
+				assert.Nil(t, resp)
+				st, ok := status.FromError(err)
+				require.True(t, ok)
+				assert.Equal(t, codes.FailedPrecondition, st.Code())
+				assert.Equal(t, backupError.Error(), st.Message())
+				require.Len(t, st.Details(), 1)
+				detailedError, ok := st.Details()[0].(*backupv1beta1.Error)
+				require.True(t, ok)
+				assert.Equal(t, tc.code, detailedError.Code)
+			})
+		}
+	})
 
-	for _, tc := range []struct {
-		testName    string
-		backupError error
-		code        backupv1beta1.ErrorCode
-	}{
-		{
-			testName:    "xtrabackup not installed",
-			backupError: backup.ErrXtrabackupNotInstalled,
-			code:        backupv1beta1.ErrorCode_ERROR_CODE_XTRABACKUP_NOT_INSTALLED,
-		},
-		{
-			testName:    "invalid xtrabackup",
-			backupError: backup.ErrInvalidXtrabackup,
-			code:        backupv1beta1.ErrorCode_ERROR_CODE_INVALID_XTRABACKUP,
-		},
-		{
-			testName:    "incompatible xtrabackup",
-			backupError: backup.ErrIncompatibleXtrabackup,
-			code:        backupv1beta1.ErrorCode_ERROR_CODE_INCOMPATIBLE_XTRABACKUP,
-		},
-	} {
-		t.Run(tc.testName, func(t *testing.T) {
-			backupError := fmt.Errorf("error: %w", tc.backupError)
-			backupService.On("PerformBackup", mock.Anything, mock.Anything).
-				Return("", backupError).Once()
+	t.Run("mongodb", func(t *testing.T) {
```

Comment:
This subtest is running inside `TestStartBackupErrors` function. Since we want to check successful running, we need to place it somewhere else.

Reply:
I think a better thing is to rename the function, better to keep tests for errors/success in the same place.